rogue/src/main.rs

```diff
@@ -289,7 +292,7 @@ impl State {
         // Build the floor layout, and update the dubug map build animation.
         self.mapgen.reset();
         let mut builder = map_builders::random_builder(depth);
-        builder.build_map();
+        let water_spawn_table = builder.build_map();
```

Comment:
This has gotten a bit awkward because it is so specific: why is the `build_map` method returning a data object encapsulating water spawn requests? I do think the design here is mostly correct, but since it is the *only* case we do this, it ends up looking bizzare.